src/types.jl

```diff
@@ -142,8 +145,18 @@ mutable struct IFeatureLayer <: AbstractFeatureLayer
     end
 end
 
-mutable struct Feature
+mutable struct Feature <: AbstractFeature
+    ptr::GDAL.OGRFeatureH
+end
+
+mutable struct IFeature <: AbstractFeature
     ptr::GDAL.OGRFeatureH
+
+    function IFeature(ptr::GDAL.OGRFeatureH = C_NULL)
+        feature = new(ptr)
+        finalizer(destroy, feature)
```

Comment:
I think we might need to be careful about the geometries associated with the feature (see https://gdal.org/api/python_gotchas.html#python-crashes-if-you-use-an-object-after-deleting-an-object-it-has-a-relationship-with, https://gdal.org/api/python_gotchas.html#certain-objects-contain-a-destroy-method-but-you-should-never-use-it).

Maybe it can be achieved by defining [`ownedby`](https://github.com/yeesian/ArchGDAL.jl/blob/7d7bba0f649981d71048c64a24f855948af7f7c3/src/types.jl#L176-L192) for the Geometries, doublechecking all the methods that relate geometries and features and making sure that they match the intended usage? (Leaving this as an open question)

Reply:
FWIW, the tests run, including those of GeoDataFrames. These tests include Features with Geometries. Furthermore, I believe that the getgeom calls create a copy (hence the improvements by using `stealgeometry` over in #266).

Reply:
From the documentation it is the other way around. A feature is owned by a layer:

> The returned feature becomes the responsibility of the caller to delete with [OGR_F_Destroy()](https://gdal.org/api/vector_c_api.html#ogr__api_8h_1a16fb8126aa932f4b4acdaca5ea99bbbb). It is critical that all features associated with an [OGRLayer](https://gdal.org/api/ogrlayer_cpp.html#classOGRLayer) (more specifically an [OGRFeatureDefn](https://gdal.org/api/ogrfeature_cpp.html#classOGRFeatureDefn)) be deleted before that layer/datasource is deleted.

So I'd say destroying the `Feature` as I do in this PR is absolutely fine, but we need to investigate what the hierarchy of owned objects is.

Reply:
Ohh okay thanks for clarifying, that makes sense! No need to hold up the rest of this PR for a hypothetical concern